packages/utils/src/logger/logger.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * This is a log4js wrapper module, in case we want to use another logging library in the future
+ * Method signature could be made more generalized if need be
+ * We NEED to use 'log4js-api' because this is an npm library that others will consume.
+ * This is to avoid dependency hell: we need to leave _them_ the responsiblity for configuring log4js.
+ * That also means that only the top package with no HOPR package dependency will use `configure` from 'lo4js'.
+ * In other words, only `hoprd` package will use 'configure', the other will only use lo4js-api dependency.
+ */
+import { getLogger, Logger as Log4jsLogger } from '@log4js-node/log4js-api'
+import { configure as log4jsConfigure } from 'log4js'
+import { Configuration as Log4jsConfiguration } from 'log4js'
+import Debug from 'debug'
+
+/** for hoprd only */
+type Configuration = Log4jsConfiguration
+function configure(config: Configuration): void {
+  log4jsConfigure(config)
+}
+
+export const setupConfigLogger = () => {
+  configure({
+    appenders: { out: { type: 'stdout' } },
+    categories: { default: { appenders: ['out'], level: 'debug' } }
+    // appenders: { custom: { type: ipfsAppender } },
+    // categories: { default: { appenders: ['custom'], level: 'debug' } }
+  })
+}
+
+export abstract class Logger {
+  static getLogger(category?: string): Logger {
+    return process.env.DEBUG ? new DebugLogger(category) : new ConfigLogger(category)
+  }
+  abstract trace(message: unknown, ...args: unknown[]): void
+  abstract fatal(message: unknown, ...args: unknown[]): void
+  abstract error(message: unknown, ...args: unknown[]): void
+  abstract warn(message: unknown, ...args: unknown[]): void
+  abstract info(message: unknown, ...args: unknown[]): void
+  abstract debug(message: unknown, ...args: unknown[]): void
+}
+class DebugLogger implements Logger {
+  private readonly logger: Debug.Debugger
+
+  public constructor(category?: string) {
+    this.logger = Debug(category)
+  }
+
+  protected log(message: unknown, ...args: unknown[]): void {
+    if (args?.length) this.logger(message, ...args)
+    else this.logger(message)
+  }
+
+  public debug(message: unknown, ...args: unknown[]): void {
+    this.logger.log = console.debug.bind(console)
```

Comment:
Why are we using console and not the logger instance?

Reply:
Yep, I think this should be the other way around.
The point would be to bind console.log to Debug logger instance here.
For log4js, it is possible too. Not sure now to do it for Debug library.

For log4js, here it goes. I quote an answer to a user in the Slack of log4js, from Gareth Jones the maintainer of lo4js:
> Instead of console.log you would have to use logger.info - or you could try something like console.log=logger.info.bind(logger) but that might not work if your HTML runs in a different process to the test where you define logger.